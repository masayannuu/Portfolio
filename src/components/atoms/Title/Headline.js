import React from 'react'
import styled from 'styled-components'
import { Title } from 'reactbulma'
import InViewMonitor from 'react-inview-monitor'

const StyledTitle = styled(Title)`
  font-feature-settings: 'palt';
  letter-spacing: 0.2em;
  margin-bottom: 95px !important;
  -webkit-animation-duration: 5s;
  -moz-animation-duration: 5s;
  animation-duration: 5s;
`

const Headline = props => (
  <InViewMonitor classNameNotInView="vis-hidden" classNameInView="animated fadeInUp">
    <StyledTitle className="is-size-3-mobile is-size-1-desktop is-size-1-tablet is-size-1-widescreen">
      {props.children}
    </StyledTitle>
  </InViewMonitor>
)

export default Headline
