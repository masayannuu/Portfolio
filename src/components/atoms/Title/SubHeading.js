import React from 'react'
import styled from 'styled-components'
import { SubTitle } from 'reactbulma'

const StyledSubTitle = styled(SubTitle)`
  font-feature-settings: 'palt';
  letter-spacing: 0.1em;
`
const SubHeading = props => <StyledSubTitle is="2">{props.children}</StyledSubTitle>

export default SubHeading
