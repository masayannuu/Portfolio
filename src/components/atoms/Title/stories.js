import React from 'react'
import { storiesOf } from '@storybook/react'
import { Headline, SubHeading } from './index'

storiesOf('Atoms.Title', module)
  .add('Headline', () => <Headline>Headline</Headline>)
  .add('SubHeading', () => <SubHeading>SubHeading</SubHeading>)
