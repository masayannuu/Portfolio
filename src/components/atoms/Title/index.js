import Headline from './Headline'
import SubHeading from './SubHeading'

export { Headline, SubHeading }
