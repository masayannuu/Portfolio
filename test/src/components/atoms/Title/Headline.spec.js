import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { Headline } from '../../../../../src/components/atoms/Title/index'

describe('Heading Headline Atoms', () => {
  it('should render Headline', () => {
    const wrapper = Enzyme.shallow(<Headline>Headline</Headline>)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
