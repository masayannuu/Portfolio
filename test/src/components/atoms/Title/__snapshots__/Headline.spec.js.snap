// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Headline Atoms should render Headline 1`] = `
<Headline__StyledTitle>
  Headline
</Headline__StyledTitle>
`;
